tests/unit/s2n_async_pkey_test.c

```diff
@@ -296,6 +341,48 @@ int main(int argc, char **argv)
             EXPECT_SUCCESS(s2n_config_free(server_config));
             EXPECT_SUCCESS(s2n_config_free(client_config));
         }
+
+        /*  Test: sign with different pkey and apply with s2n_connection having different chain_and_key */
+        {
+
+            struct s2n_config *server_config, *client_config;
+            EXPECT_NOT_NULL(server_config = s2n_config_new());
+            EXPECT_SUCCESS(s2n_config_set_async_pkey_validation_mode(server_config, S2N_ASYNC_PKEY_VALIDATION_STRICT));
+            EXPECT_SUCCESS(s2n_config_add_cert_chain_and_key_to_store(server_config, chain_and_key));
```

Comment:
Should you test that the handshake is possible with the bad signature if S2N_ASYNC_PKEY_VALIDATION_STRICT isn't set?

Reply:
Let me add it as 1 more test case